backend/dataall/modules/datasets_base/db/dataset_models.py

```diff
@@ -141,3 +141,23 @@ class Dataset(Resource, Base):
     @classmethod
     def uri(cls):
         return cls.datasetUri
+
```

Comment:
I think that is a great idea! It would definitely allow more flexibility, scalability, and ease of analysis. We can have a Dataset Table (Fact Table) containing the core dataset metadata such as dataset uri, name, aws account, region, etc. We can then have other dimension tables around it and keep adding them as we expand our dataset properties. We will have to think of the design more deeply though. We might want to think of indexign and partitioning strategies to optimize performance too.